aquarium/aquarium.js

```diff
@@ -2136,38 +2174,50 @@ $(function(){
     return path.substring(0, path.lastIndexOf('/'));
   }
 
-  function onPresentChange() {
-    // When we begin or end presenting, the canvas should be resized
-    // to the recommended dimensions for the display.
-    resize();
-
-    g_shadersNeedUpdate = true;
-
-    if (g_vrDisplay.isPresenting) {
-      if (g_vrDisplay.capabilities.hasExternalDisplay) {
-        removeButton(vrButton);
-        vrButton = addButton("Exit VR", "E", getCurrentUrl() + "/vr_assets/button.png", onExitPresent);
-      }
-    } else {
-      if (g_vrDisplay.capabilities.hasExternalDisplay) {
-        removeButton(vrButton);
-        vrButton = addButton("Enter VR", "E", getCurrentUrl() + "/vr_assets/button.png", onRequestPresent);
-      }
-    }
-  }
+  vrButtonURL = getCurrentUrl() + "/../aquarium-vr/vr_assets/button.png";
 
   function onRequestPresent() {
-    g_vrDisplay.requestPresent([{ source: canvas }]).then(function() {}, function() {
-      console.error("request present failed.");
+    return navigator.xr.requestSession('immersive-vr').then((session) => {
+        removeButton(vrButton);
+        vrButton = addButton("Exit VR", "E", vrButtonURL, onExitPresent);
+        session.isImmersive = true;
+        g_session = session;
+
+        session.updateRenderState({ baseLayer: new XRWebGLLayer(session, gl) });
+
+        let refSpaceType = 'local';
```

Comment:
I suspect that this will eventually want to be changed from `'local'` to `'local-floor'` to get some of the effect I was talking about with aligning to the user's floor. It probably won't make much visual difference at the moment, though, given the scale issues.